lightning/src/chain/chaininterface.rs

```diff
@@ -19,6 +19,7 @@ use std::sync::atomic::{AtomicUsize, Ordering};
 use std::collections::HashSet;
 use std::ops::Deref;
 use std::marker::PhantomData;
+use std::ptr::eq as ptr_eq;
```

Comment:
Maybe just use std::ptr and refer to things in the file as ptr::eq? It should avoid any confusion of it being a different function.